compiler/src/dotty/tools/dotc/ast/Desugar.scala

```diff
@@ -1943,12 +1943,26 @@ object desugar {
       case AndType(tp1, tp2) => stripToCore(tp1) ::: stripToCore(tp2)
       case _ => defn.AnyType :: Nil
     }
+
+    val refinements1 = refinements.flatMap {
+      case tree: ValDef if tree.mods.is(Mutable) =>
+        val getter =
+          cpy.DefDef(tree)(name = tree.name, paramss = Nil, tpt = tree.tpt, rhs = tree.rhs)
+            .withMods(Modifiers(tree.mods.flags & (AccessFlags | Synthetic), tree.mods.privateWithin))
```

Comment:
This expression appears twice and could be lifted out. Also, one can use `withFlags`.
```scala
val mods1 = tree.mods.withFlags(tree.mods.flags & AccessFlags | Synthetic)
```